services/cell-logger/cell-logger

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+import dbus
+import re
+import time
+import csv
+
+#Initialize DBus system bus 
+
+bus = dbus.SystemBus()
+
+#Query modem data using the ModemManager DBus object. 
+
+def get_modem_rssi():
+
+    modem_signal_iface = 'org.freedesktop.ModemManager1.Modem.Signal'
+
+    modem_manager_object = bus.get_object('org.freedesktop.ModemManager1', 
+                                          '/org/freedesktop/ModemManager1')
+    modem_manager_iface = dbus.Interface(modem_manager_object, 
+                                         'org.freedesktop.DBus.ObjectManager'
+
+    modem_data = modem_manager_iface.GetManagedObjects()
+
+#Extract modem path from the DBus data dictionary
+
+    modem_path = list(modem_data.keys())[0]    
+
+#Check if the 'Lte' dictionary has signal values assigned. If no values are 
+#found, assume the modem is connected via 'Umts'
+
+    if(list(modem_data[dbus.ObjectPath(modem_path)][
+                       dbus.String(modem_signal_iface)][
+                           dbus.String('Lte')].keys())):
+        
+        cell_tech = 'Lte'
+
+    else:
+
+        cell_tech = 'Umts'
+
+#Get RSSI data from the dictionary. If there is still no data, assume the modem 
+#ID has changed and needs to have the signal info update rate set. 
+
+    try:
+        
+        rssi = modem_data[dbus.ObjectPath(modem_path)][
+                          dbus.String(modem_signal_iface)][
+                          dbus.String(cell_tech)][dbus.String('rssi')]
+        
+    except(KeyError): 
+
+        set_update_rate(modem_path)
+
+        return ''
+
+    return [int(float(rssi)),cell_tech]
+
+#Assign a signal quality retrieval refresh rate of 1 second using the 'Setup()'
+#method of the 'Signal' object for the modem in modem_path.
+
+def set_update_rate(modem_path):
+
+    modem_object = bus.get_object('org.freedesktop.ModemManager1', modem_path)
+    modem_iface = dbus.Interface(modem_object,
+                                 'org.freedesktop.ModemManager1.Modem.Signal')
+    modem_iface.Setup(dbus.UInt32(1))
+
+while(True):
+
+    timestamp = int(time.time()) 
+
+    [signal, cell_tech] = get_modem_rssi()
+    
+    if (signal != ''):
+
+        with open('/data/log/cell.csv', mode = 'a') as log:
```

Comment:
Does this `fopen` and `fclose` for every loop?

Reply:
Yes, every time the code inside the _with_ statement is ran, all the resources used will be closed. This is likely to change (and eventually disappear) after a transition to a local postgres database logging.